client/src/components/graph/graph.tsx

```diff
@@ -275,14 +257,17 @@ class Graph extends React.Component<GraphProps, GraphState> {
       camera: null,
       modelTF,
       modelInvTF: mat3.invert(mat3.create(), modelTF),
-      projectionTF: createProjectionTF(viewport.width, viewport.height),
+      projectionTF: createProjectionTF({
```

Comment:
we now need to take spatial mode into account, because non-spatial embedding without 32px padding would actually go beyond the viewport, which is not good. So we only use 0 paddings when it's spatial mode